### PR TITLE
phase1: fix leaking targets between branches

### PR DESCRIPTION
**Pull request by ynezz**
@robimarko noticed, that after rename of `ipq807x` target in main branch to `qualcommax/ipq807x` subtarget, that buildbot is trying to build this new `qualcommax/ipq807x` subtarget on `openwrt-23.05` branch as well.
    
@f00b4r0  [later explained](https://github.com/openwrt/buildbot/issues/14#issuecomment-1595839703), that this is by design, his initial idea was to find exhaustive list of all targets and let the `checkarch` step do the final triaging.
    
I find this approach confusing, because if the subtarget is not present in that branch, we shouldn't have builder for it configured as well. Furthermore wasting roughly 5 minutes of precious buildworker time to checkout all feeds and then just find out, that we're not going to use those seems suboptimal as well.
    
So lets fix it by using builders for targets as present in respective branches.

Fixes: #14

### CODE REVIEW

**Comment by robimarko** [comment]
To me this seems like a better approach than having builders that get triggered to then later fail despite that being intended, even worse as they are using resources.

File: phase1/master.cfg [hunk 1]
```diff
@@ -734,8 +735,7 @@ c["builders"].append(
 )
 
 
-# NB the phase1 build factory assumes workers are single-build only
```
Comment (f00b4r0) [inline phase1/master.cfg:652]: I wouldn't move this comment as it is essential to bear it in mind when editing build steps in the factory.
Moving it down below makes it "invisible" in that context and it doesn't apply to the builder list preparation.

**Review by f00b4r0 — COMMENTED** [review]
LGTM, with the comment caveat. Definitely more efficient.
So now the checkarch step is moot, although it probably doesn't hurt to leave it there just to catch unforeseen issues down the road.

File: .github/workflows/build-push.yml [hunk 2]
```diff
@@ -39,9 +39,8 @@ jobs:
       - name: Lint with ruff
         run: ruff phase*/master.cfg
 
-# FIXME
-#     - name: Stylecheck with black
-#       run: black phase*/master.cfg
+      - name: Stylecheck with black
+        run: black phase1/master.cfg
```
Comment (aparcar) [inline .github/workflows/build-push.yml:43]: 🥇

File: phase1/master.cfg [hunk 3]
```diff
@@ -1037,6 +1037,9 @@ def prepareFactory(target):
             descriptionDone="dl/ pruned",
             command="find dl/ -mindepth 1 -atime +15 -delete -print",
```
Comment (f00b4r0) [inline phase1/master.cfg:1038]: missed the -print -delete reorder :)

Comment (ynezz) [reply phase1/master.cfg:1038]: I've noticed that remark, so I've tried it locally, didn't spot any difference with the output, so simply skipped that.

**Pull request merged**